monitoring.adoc

```diff
@@ -777,17 +777,68 @@ account activity for one user without filling up mailbox.log with log
+* Zimbra context
+* Component name (e.g. `soap`, `mailbox`, `mbxmgr`, etc)
+* Log message
+
+You can read more about Log4j patterns in https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html[Log4j PatternLayout documentation].
+
+*Thread prefix in mailbox.log*
+
+Thread numbers in mailbox.log are prefixed to identify internal components. The following prefixes are currently used in {product-name}: `btpool`, `pool`, `LmtpServer`, `ImapServer`,`ImapSSLServer`, `Pop3Server`, `Pop3SSLServer`, `ScheduledTask`, `Timer`, `AnonymousIoService`, `CloudRoutingReaderThread`, `GC`, `SocketAcceptor`, `Thread`, `qtp`.
+
+*Zimbra Context in mailbox.log*
+
+`[%z]` section in the log pattern describes Zimbra context and consists of key-value pairs in the format `key=value`, separated by semi-colons (`;`). The following key value pairs are currently supported and may be recorded in log entries in any order and eny combination:
+
+* `ip` - IP of the TCP/IP client making a request
+* `oip` - originating IP address. When a request is made through NGINX proxy, this value will contain the IP address of the client application, while `ip` value will contain the IP address of the proxy server
+* `cid` - connection id of a server that is monotonically increasing - useful for tracking individual connections
+* `id` - ID of the target account
+* `name` - name of the target account (email address)
+* `aid` - ID of the authenticated account. Only present if target account is different then authenticated account
+* `aname` - name of the authenticated account. Only present if target account is different then authenticated account
+* `mid` - Id of requested mailbox. Only present if request is dealing with a mailbox
+* `ua`  - name of the client application (i.e. User Agent)
+* `via` - list of IP addresses and user-agents of the request's proxy chain
+* `soapId` - ID assigned to a SOAP request to track proxied hops for a particular request
+* `msgid` - value of Message-ID header of the message being operated on
+*  `ds` - name of the Data Source being operated on
+* `port` - server port to which the client connected
+* `oport` - originating port number of request
+* `oproto` - originating protocol of request. This can be passed by internal components that make SOAP requests on behalf of a user (e.g. MTA and IMAPd)
+
+The example below is a record showing that
+on October 25, 2007, 28 minutes after midnight, a POP3 client with IP address `222.173.186.17` has contacted the {product-name} server and that the request was proxied through a local proxy server with IP 10.1.1.136.
+
+image:images/poplogmessage.png[Mailbox Log Entry for POP]
+
+Log records reported by threads that serve SOAP requests will usually contain URL of the request being served, as in the following example:
+
+image:images/zwclogmessage.png[Mailbox Log Entry for SOAP]
```

Comment:
I think the above section as things currently are should get moved up to the thread context section and out of the zimbra context section since it is about the thread and not zimbra.

Having said that I'm wondering why we have the url in the thread context and not in the zimbra context. It doesn't really seem like a thread attribute I expect it would change for a given thread depending on the zimbra context although I haven't ran across examples of the url in my data yet investigate.

Reply:
Our document formatting is not real clear so I could be confused on the above (indentation and or section numbers) may help perhaps this all happens up stream since I think I'm mostly looking at a raw file.

But if we have
1. Review mailbox.log Records
   1. Log pattern
   2. Thread prefix in mailbox.log
   3. {product name} Context in mailbox.log
      all the text

Then the stuff in all the text i think should get split up better in 1, 2, and 3. But if we have

1. ...
   1. ...
   2. ...
   3. ...
      three text

more text for one

where all the text is really a part  of Review mailbox.log Records then it may be OK although some of that still could probably go to the different sections.